pipit/trace.py

```diff
@@ -651,3 +651,39 @@ def calc_exc_time_in_bin(events):
         df.insert(0, "bin_end", edges[1:])
 
         return df
+
+
+def detect_pattern(
+    self, data, iterations=None, window_size=None, process=0, metric="time.exc"
+):
+    import stumpy
+
+    y = data.loc[(data["time.exc"].notnull()) & (data["Process"] == 0)][
```

Comment:
should probably check if the metric argument exists in the dataframe before this function, and if not, run the appropriate function

Reply:
same thing with time.exc, which I'm assuming we use for the matrix profile and is different from what the metric is?

Reply:
also is there a reason why we use time.exc for the matrix profile?